tests/puppeteer/campaigns/data/faker/address.js

```diff
@@ -0,0 +1,22 @@
+const faker = require('faker');
+
+const {countries} = require('@data/demo/countries');
+
+module.exports = class Address {
+  constructor(addressToCreate = {}) {
+    this.email = addressToCreate.email || faker.internet.email(this.firstName, this.lastName, 'prestashop.com');
```

Comment:
```suggestion
    this.email = addressToCreate.email !== undefined ? faker.internet.email(this.firstName, this.lastName, 'prestashop.com') : addressToCreate.email;
```
So after you can use the `null` value, otherwise if you want to do the same thing for `firstName`, `lastName`, `postalCode`, ... you'll add an infinite of parameter to the `createEditAddress` method.

Reply:
When creating an email address, we should use an existing email, then we should add it in parameter -> `const createAddressData = new AddressFaker({email: 'pub@prestashop.com', country: 'France'});`
When editing the customer email address, the email field cannot be edited...then the condition should be in the `createEditAddress` method to not set an email -> if (setEmailValue) await this.setValue(this.customerEmailInput, addressData.email);
I think it's the best solution
What do you think ?

Reply:
with my check you can do:
```js
createAddressData = new AddressFaker({email: 'pub@prestashop.com', country: 'France'});
```

Create faker email
```js
createAddressData = new AddressFaker({country: 'France'}); 
```
Create null email
```js
createAddressData = new AddressFaker({email: null, country: 'France'});
```
Create empty email
```js
createAddressData = new AddressFaker({email: '', country: 'France'});
```
And the `if (setEmailValue)` becomes obsolete

Reply:
Not necessary anymore because of this check `await this.elementVisible(this.customerEmailInput, 2000))`